peakina/datasource.py

```diff
@@ -44,15 +48,33 @@ class DataSource:
     reader_kwargs: dict[str, Any] = field(default_factory=dict)
     fetcher_kwargs: dict[str, Any] = field(default_factory=dict)
 
-    def __post_init_post_parse__(self) -> None:
-        self._fetcher: Fetcher | None = None
-        self.scheme = urlparse(self.uri).scheme
-        if self.scheme not in PD_VALID_URLS:
-            raise AttributeError(f"Invalid scheme {self.scheme!r}")
+    # TODO: This is temporary, in the future we will only support V2
+    # and get rid of this condition + update the CI (link/test)
+    if _PYDANTIC_VERSION_ONE is True:
+
+        def __post_init_post_parse__(self) -> None:
+            self._fetcher: Fetcher | None = None
+            self.scheme = urlparse(self.uri).scheme
+            if self.scheme not in PD_VALID_URLS:
+                raise AttributeError(f"Invalid scheme {self.scheme!r}")
+
+            self.type = self.type or detect_type(urlparse(self.uri).path, is_regex=bool(self.match))
+
+            validate_kwargs(self.reader_kwargs, self.type)
+
+    else:
+
+        def __post_init__(self) -> None:
+            self._fetcher: Fetcher | None = None  # type: ignore[no-redef]
+            self.scheme = urlparse(self.uri).scheme
+            if self.scheme not in PD_VALID_URLS:
+                raise AttributeError(f"Invalid scheme {self.scheme!r}")
+
+            self.type = self.type or detect_type(urlparse(self.uri).path, is_regex=bool(self.match))
 
-        self.type = self.type or detect_type(urlparse(self.uri).path, is_regex=bool(self.match))
+            validate_kwargs(self.reader_kwargs, self.type)
 
-        validate_kwargs(self.reader_kwargs, self.type)
+    model_config = ConfigDict(arbitrary_types_allowed=True)
```

Comment:
this is only for v2 right? (hence should be in the `else` block)

Reply:
good catch, thanks !